docs/guides/tutorials/build-a-graphql-webservice.md

```diff
@@ -0,0 +1,184 @@
+
+The corresponding source code for this tutorial is available on [GitHub](https://github.com/zio/zio-quickstart-graphql-webservice). If you haven't read the [ZIO Quickstart: Building GraphQL Web Service](../quickstarts/graphql-webservice.md) yet, we recommend you to read it first and download and run the source code, before reading this tutorial.
+
+## Installation
+
+In this tutorial, we will build a GraphQL API using _Caliban_, and in order to serve it, we will use the _ZIO HTTP_ library. So let's install the necessary dependencies by putting the following lines in the _build.sbt_ file:
+
+```sbt
+libraryDependencies ++= Seq(
+  "dev.zio"               %% "zio"              % "2.0.0",
+  "com.github.ghostdogpr" %% "caliban"          % "2.0.0",
+  "com.github.ghostdogpr" %% "caliban-zio-http" % "2.0.0",
+  "io.d11"                %% "zhttp"            % "2.0.0-RC10"
+)
+```
+
+Now, we are ready to jump into the next section.
+
+## Creating a GraphQL Schema
+
+When working with GraphQL, we need to define a _schema_, which defines what kind of data with which types we can query. In GraphQL, schemas are defined in their type system. For example, assume we have a data type called _Employee_, which had defined in scala as the following:
+
+```scala mdoc:silent
+case class Employee(name: String, role: Role)
+
+sealed trait Role
+object Role {
+  case object SoftwareDeveloper       extends Role
+  case object SiteReliabilityEngineer extends Role
+  case object DevOps                  extends Role
+}
+```
+
+It has two fields, _name_, and _role_, and the _role_ field is a _sealed trait_, which means that it can only be one of the three values, _SoftwareDeveloper_, _SiteReliabilityEngineer_, or _DevOps_.
+
+If we want to define such a data type in GraphQL, we have something like this:
+
+```graphql
+type Employee {
+  name: String!
+  role: Role!
+}
+
+enum Role {
+  SoftwareDeveloper
+  SiteReliabilityEngineer
+  DevOps
+}
+```
+
+After defining the schema for our data types, the next step is to define queries that can be performed on the data. For example, if we want to query all the employees with a specific role, we can do that like this:
+
+```graphql
+type Queries {
+  employees(role: Role!): [Employee!]!
+}
+```
+
+Similarly, if we want to query an employee by its name, we can define that like this:
+
+```graphql
+type Queries {
+  employee(name: String!): Employee
+}
+```
+
+Fortunately, we are not required to define the schema in GraphQL manually. Instead, we can use the _Caliban_ library which has a built-in facility to generate the schema from our data types defined in Scala:
+
+```scala mdoc:silent
+import caliban.schema.Annotations.GQLDescription
+
+case class EmployeesArgs(role: Role)
+case class EmployeeArgs(name: String)
+
+case class Queries(
+    employees: EmployeesArgs => List[Employee],
+    employee: EmployeeArgs => Option[Employee]
+)
+```
+
```

Comment:
Maybe quickly mention that fields can return a `ZIO` as well (otherwise ZIO is barely used at all 😄 ).